docs/architecture/collaborative-storage.md

```diff
@@ -0,0 +1,109 @@
+
+On POSIX, each space could for example be mapped to it's own directory in a special spaces folder under the oCIS root folder.
+
+## ID to Path Lookup
+
+Infinite Scale uses file IDs to efficiently identify files within a file tree. The lookup from a given ID to a path within the oCIS file tree is a very basic function that more or less defines the Infinite Scale performance. The functionality to for example query the file path for a given Inode number (which is the nearest equivalent for the Infinite Scale file ID) can not be done with standard POSIX system calls.
+
+The interface defining the collaborative storage needs an abstraction for this particular function, returning the file id for a given path, and returning the path for given id.
+
+## Change Notification
+
+When a file is changed by a process outside of oCIS, this needs to be monitored by oCIS to quickly maintain internal caches and data structures as required.
+
+The collaborative storage driver needs a way to achieve that. The easiest way for an POSIX based collaborative storage is inotify, that needs to be set up recursively on a file tree to record changes. Additional it is a challenge to destinguish between changes that were done from external activity and the ones that oCIS creates by its own file operations.
+
+For GPFS, there should be an API function delivering that. [TBD]
+
+## ETag Propagation
+
+ownCloud requires that changes which happen "down" in a tree, can be detected in the root element of the tree. That happens through the change of the ETag metadata of each file and/or directory. An ETag is a random, text based tag, that only has one requirement: It has to change its content if a resource further down in the file tree has changed either its content or its metadata. (See [this issue](https://github.com/owncloud/ocis/issues/3782) for further discussion about the ETag/CTag).
+
+POSIX file systems do not maintain a change flag like the ETag by default. The file time stamps (atime, ctime, mtime) in general are not fine granular enough (only seconds for some file systems) and depend on the server time, which renders them useless in a distributed environment.
+
+Infinite Scale needs to implement ETag propagation "up". For the collaborative storage, that needs to be combined with the change notification described above.
+
+Certain file systems implement this functionality either independently from Infinite Scale (EOS) or at least support proper change notifications (Ceph, GPFS?).
+
+## Metadata Management
+
+Metadata are data "snippets" that are as tightly attached to files as ever possible. In best case, a rename of a file silently keeps the metadata as well. In POSIX, this can be achieved by extended file attributes with certain limitations.
```

Comment:
`...by extended file attributes...`
hmmm, havn't we kicked out extended file attributes with ocis v3 and we use from then on messagepack by default? Also see the prerequisites in the admin docs as linked in the comment before.

Reply:
the text is about collaborative storage where it is even more important that the metadata is attached to the file because there can be file system manipulation outside of oCIS' control.

If we can not use xattrs, we maybe can store a "link" to the messagepack file in xattrs - so that we can find the metadata even if the file was moved. Needs more investigation.